framework/src/modules/chain/modules/blocks/process.js

```diff
@@ -539,66 +548,36 @@ Process.prototype.loadBlocksFromPeer = function(peer, cb) {
  */
 Process.prototype.generateBlock = function(keypair, timestamp, cb) {
 	// Get transactions that will be included in block
-	const transactions = modules.transactions.getUnconfirmedTransactionList(
-		false,
-		MAX_TRANSACTIONS_PER_BLOCK
-	);
-	const ready = [];
-
-	async.eachSeries(
-		transactions,
-		(transaction, eachSeriesCb) => {
-			modules.accounts.getAccount(
-				{ publicKey: transaction.senderPublicKey },
-				(err, sender) => {
-					if (err || !sender) {
-						return setImmediate(eachSeriesCb, 'Sender not found');
-					}
-
-					// Check transaction depends on type
-					if (library.logic.transaction.ready(transaction, sender)) {
-						// Verify transaction
-						return library.logic.transaction.verify(
-							transaction,
-							sender,
-							null,
-							true,
-							verifyErr => {
-								if (!verifyErr) {
-									ready.push(transaction);
-								}
-								return setImmediate(eachSeriesCb);
-							},
-							null
-						);
-					}
-					return setImmediate(eachSeriesCb);
-				}
+	const transactions =
+		modules.transactions.getUnconfirmedTransactionList(
+			false,
+			MAX_TRANSACTIONS_PER_BLOCK
+		) || [];
+
+	modules.processTransactions
+		.verifyTransactions(transactions)
+		.then(responses => {
+			const readyTransactions = transactions.filter(tx =>
+				responses
+					.filter(response => response.status === TransactionStatus.OK)
+					.map(r => r.id)
```

Comment:
Please be consistent. Use `transactions` keyword instead of `tx` and response instead of `r`.